### PR TITLE
Adding in support for passing in an access token

### PR DESCRIPTION
**Pull request by mannyjimenez0810**

### CODE REVIEW

File: lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb [hunk 1]
```diff
@@ -31,7 +31,7 @@ def self.run(params)
         UI.user_error!("Couldn't find binary at path #{binary_path}") unless File.exist?(binary_path)
         binary_type = binary_type_from_path(binary_path)
 
-        auth_token = fetch_auth_token(
+        auth_token = params[:access_token] || fetch_auth_token(
```
Comment (lfkellogg) [inline lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb:34]: The `access_token` param does not exist yet. You'd have to add it to `self.available_options` below.

Comment (mannyjimenez0810) [reply lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb:34]: Thanks!

**Review by tagboola — CHANGES_REQUESTED** [review]
Hate to be the wet blanket here, but I think we this would require an API review. What worries me is that this is not a parameter that the Firebase CLI supports, even though it _seemingly_ supports WIF. Ideally, we want a consistent auth experience across _all_ CLI tools.

https://github.com/firebase/firebase-tools/issues/3926
https://tables.area120.google.com/table/8Ls4BVBK9Fn98JVbhvK_jQ/row/9CaYXMOtErA4El6aLw__6A

**Comment by mannyjimenez0810** [comment]
The API review is unfortunate. Did we go through an initial review for the service_credentials_file? I tried looking for a api doc, but couldn't find one for that I found one for other parts.

Also why should we wait for Firebase CLI? If google support this auth shouldn't we also support this auth.  Looks like they updated their ruby auth to support this not too long ago https://github.com/googleapis/google-auth-library-ruby/issues/346. I did see some users also want Fastlane to have this feature, although that work was started months ago, it has not been picked up recently.

**Comment by tagboola** [comment]
If the real problem is we need to support authentication with Workload Identity Federation, which is referenced in #274, I think we should try to do it in a way that's consistent with other clients.  Based on https://google.aip.dev/auth/4117, the guidance for authenticating with Workload Identity Federation is to generate a JSON configuration file that can then be used with [Application Default Credentials](https://google.aip.dev/auth/4110). The plugin already [supports](https://github.com/fastlane/fastlane-plugin-firebase_app_distribution/blob/b2dd465d20f739edef9f66093298c425525f2c9a/lib/fastlane/plugin/firebase_app_distribution/helper/firebase_app_distribution_auth_client.rb#L35) Application Default Credentials, so why not try and support this way?

**Comment by mannyjimenez0810** [comment]
@tagboola and I chatted offline. Unless this is customer blocking we will not pursue this any longer unless other CLI tools also update to support this authentication method. Happy to chat @lfkellogg if you had some thoughts or wanted some more context on this.

**Pull request closed without merging**